### DISCUSSION TITLE
UpsertOne with composite key

### DISCUSSION THREAD

**Discussion by nettrino** [prisma/prisma-client-go#954] [Q&A] [answered]
Apologies if this is the docs and I've missed it or whether this warrants a ticket instead, I'm trying to use UpsertOne for a schema with a composite key and get a too many arguments error. For instance, I want to call upsert one for the schema below

```prisma

enum HostingPlatform {
    pA
    pB
    unknown
}

model TestPlatform {
    platformId   String
    name         String
    url          String
    createdAt    DateTime                            @default(now()) @map("created_at")
    updatedAt    DateTime                            @default(now()) @updatedAt @map("updated_at")
    platform     HostingPlatform

    @@id(name: "testId", [platform, platformId])
    @@map("repository_organizations")
}
```

**Comment by nettrino** [comment]
Likewise - using `Link` with composite keys upon a `CreateOne` seems to only allow a single field

**Reply by steebchen** [reply]
Can you share some code on what you are trying to do right now?

**Reply by steebchen** [reply]
I think there are actually no docs :/

I will try to add them soon via https://github.com/steebchen/prisma-client-go/issues/955). 

Basically, you can define a @@unique on your model:


```prisma
model Post {
  // ...
  @@unique([id, kind], name: "unique_id_kind")
}
```

And then query for it using the specified name (defaults to the concatenated field names):

```go
user, err := r.db.User.UpsertOne(
	db.User.UniqueIdKind(
		db.User.ID.Equals(id),
		db.User.Kind.Equals(kind),
	),
).Create(...).Update(...).Exec(ctx)
```

**Reply by nettrino** [reply]
Thank you for the very prompt reply. You mean define `@@unique` additionally to `@@id`? Because it is already implied if the id is composite that they are unique

**Reply by steebchen** [reply]
@@id should also work, yes

**Reply by nettrino** [reply]
would that be via a `EqualsUniqueWhereParam`? How would an "and" clause for two properties work for the example above ?

**Reply by nettrino** [reply]
btw not sure if https://github.com/steebchen/prisma-client-go/issues/956 is related but I can't get this to show neither with `@@id` nor addition of `@@unique`

**Reply by steebchen** [reply]
Do you not have the method `db.TestPlatform.TestID` (as `testId` is your @@id name)? If not then it is probably a bug or a missing feature

**Reply by steebchen** [reply]
Thanks for creating the other issue, I'll take a look later...

**Reply by nettrino** [reply]
> Do you not have the method db.TestPlatform.TestID (as testId is your @@id name)? If not then it is probably a bug or a missing feature

I don't - could you double check on your end using the provided snippet in #956? perhaps its a go version mismatch

**Reply by steebchen** [reply]
It was a bug, please see https://github.com/steebchen/prisma-client-go/issues/956

**Comment by steebchen** [comment, marked as answer]
This was actually a bug and is tracked via https://github.com/steebchen/prisma-client-go/issues/963
